### PR TITLE
Add issue and PR templates

### PR DESCRIPTION
**Pull request by janmeier**
Personally I love the idea of an issue template. Some people over at https://github.com/rust-lang/rust/pull/31732 are less excited than me, but personally I don't mind turning a few users away if they can't take the time to write a proper description of the issue - changes are the issue wouldn't have been fixed anyway because it wasn't explained properly :)

https://github.com/blog/2111-issue-and-pull-request-templates

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## Checklist
+- [] Link to existing issue (Closes #[issue]) or a description of the issue your are solving
+- [] All existing tests pass
+- [] You have added new tests to prevent regressions
+- [] An entry under 'Next' in the changelog
```
Comment (sushantdhiman) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can we change `Next` to `Future` :smile:

**Comment by sushantdhiman** [comment]
Some PR will not require tests, others will not require change log, Can we somehow mention that in PR template.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (janmeier) [inline .github/PULL_REQUEST_TEMPLATE.md]: Sure :)

**Comment by janmeier** [comment]
@sushantdhiman Yeah - Maybe something along the lines of 'If your PR includes changes to library code remember the following'

**Comment by sushantdhiman** [comment]
@janmeier we can use [Node.JS project](https://github.com/nodejs/node/blob/master/.github/PULL_REQUEST_TEMPLATE.md) format for PR templates

**Comment by janmeier** [comment]
Good idea @sushantdhiman - I added some changes inspired by the node project

**Pull request merged**